stdlib/public/Concurrency/Task.cpp

```diff
@@ -21,51 +21,150 @@
 #include "TaskPrivate.h"
 
 using namespace swift;
+using FutureFragment = AsyncTask::FutureFragment;
+
+void FutureFragment::destroy() {
+  auto queueHead = waitQueue.load(std::memory_order_acquire);
+  switch (queueHead.getStatus()) {
+  case Status::Executing:
+    assert(false && "destroying a task that never completed");
+
+  case Status::Success:
+    resultType->vw_destroy(getStoragePtr());
+    break;
+
+  case Status::Error:
+    swift_unknownObjectRelease(reinterpret_cast<OpaqueValue *>(getError()));
```

Comment:
Is this the standard way we release errors in the runtime?  Shouldn't we have an `swift_errorRelease` we can unconditionally call, even if it just compiles to `swift_release` in some configurations?  Is it not important that we call `swift_errorRelease` in the configurations that provide it?

Reply:
This is the standard way when we don't know whether we'll have a native Swift object or not. `swift_errorRelease` is implemented as `objc_release` where it's implemented.